cmd/collector/app/handler/otlp_receiver_test.go

```diff
@@ -88,16 +97,28 @@ func TestConsumerDelegate(t *testing.T) {
 func TestStartOtlpReceiver_Error(t *testing.T) {
 	spanProcessor := &mockSpanProcessor{}
 	logger, _ := testutils.NewLogger()
-	_, err := StartOtelReceiver(
-		OtelReceiverOptions{
-			GRPCHostPort: ":-1",
-			HTTPHostPort: ":-1",
-		},
-		logger,
-		spanProcessor,
-	)
-	assert.Error(t, err)
+	opts := optionsWithPorts(":-1")
+	_, err := StartOTLPReceiver(opts, logger, spanProcessor)
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "could not start the OTLP receiver")
+
+	newTraces := func(_ consumer.ConsumeTracesFunc, _ ...consumer.Option) (consumer.Traces, error) {
+		return nil, errors.New("mock error")
+	}
+	f := otlpreceiver.NewFactory()
+	_, err = startOTLPReceiver(opts, logger, spanProcessor, f, newTraces, f.CreateTracesReceiver)
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "could not create the OTLP consumer")
+
+	createTracesReceiver := func(_ context.Context, _ component.ReceiverCreateSettings,
+		_ config.Receiver, _ consumer.Traces) (component.TracesReceiver, error) {
```

Comment:
nit: Since none of the params are used, they can be removed altogether; it's not necessary to add the blank identifiers. Same with above `newTraces := func(...)`.
```suggestion
	createTracesReceiver := func(context.Context, component.ReceiverCreateSettings,
		config.Receiver, consumer.Traces) (component.TracesReceiver, error) {
```